### PR TITLE
Create and test MovingAverageService

### LINKED ISSUE #145: Calculate wait time as an exponential moving average
Given a backlog of call requests to process, we would like to be able to give the best possible wait time estimate to the end user. We will implement the wait time calculation as an [exponential moving average (EMA)](https://www.investopedia.com/terms/e/ema.asp) which is calculated with each API call to the `/reservations/search/startNextCall` endpoint.

The rough implementation should be as follows:

1. Search Mongo for the existing EMA.
2. If no existing EMA, use the difference between the current time and the created time for the reservation to be handled.
3. If there does exist an EMA, calculate the new EMA according the the formula linked above. The smoothing factor will be set to 2 by default in application properties.
4. The new EMA is stored on the reservation to reflect the calculated EMA at the time that this call was handled.

NOTE: the EMA stored on the reservation is *not* the same as the wait time that was shown to that reservation's end user. Rather, the `GET /reservation/{id}` endpoint will determine wait time by searching Mongo for the document most recently updated with an EMA.

### PR DESCRIPTION
**Pull request by egracer**
Fixes #145.

Refactors the exponential moving average logic from #161 into a separate service and implements unit tests to verify the behavior.

### CODE REVIEW

File: callmeback/backend/src/test/java/org/google/callmeback/api/services/MovingAverageServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    Reservation res = TestHelpers.createReservation();
+    Exception nullEx =
+        assertThrows(
+            IllegalArgumentException.class,
+            () -> {
+              movingAverageService.getNewExponentialMovingAverage(res, null, null);
+            });
+
+    assertEquals(
+        "A non-null ReservationEvent of type CONNECTED must be provided to calculate a "
```
Comment (kebrahim) [inline callmeback/backend/src/test/java/org/google/callmeback/api/services/MovingAverageServiceTest.java:54]: looks like you're using column width of 80 here. you should be using 100 for java files - go/java-style#s4.4-column-limit

File: callmeback/backend/src/test/java/org/google/callmeback/api/services/MovingAverageServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    return getMovingAverage(OptionalDouble.empty());
+  }
+
+  /**
+   * Returns a twenty-minute moving average by default. If passed a non-null previousAverage, will
+   * calculate the updated moving average based on the previous value.
+   */
+  private double getMovingAverage(OptionalDouble previousAverage) {
+    ZonedDateTime d = ZonedDateTime.of(2020, 05, 28, 10, 0, 0, 0, ZoneOffset.UTC);
+    ZonedDateTime d2 = d.plusMinutes(20);
```
Comment (kebrahim) [inline callmeback/backend/src/test/java/org/google/callmeback/api/services/MovingAverageServiceTest.java:129]: mind passing the 20 minutes in as a param to this method? it's somewhat confusing that this returns a movingAverage of 20 by default

File: callmeback/backend/src/test/java/org/google/callmeback/TestHelpers.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+import java.util.Date;
+import java.util.Optional;
+import org.assertj.core.util.Lists;
+import org.google.callmeback.api.Reservation;
+import org.google.callmeback.api.ReservationEvent;
+
+public class TestHelpers {
+
+  static final String defaultTopic = "Business";
```
Comment (kebrahim) [inline callmeback/backend/src/test/java/org/google/callmeback/TestHelpers.java:11]: this variable name should use all caps (DEFAULT_TOPIC) - go/java-style#s5.2.4-constant-names

File: callmeback/backend/src/test/java/org/google/callmeback/api/services/MovingAverageServiceTest.java (hunk 1 above)
Comment (egracer) [reply callmeback/backend/src/test/java/org/google/callmeback/api/services/MovingAverageServiceTest.java:54]: Confirmed that the column width is 100 -- this string is just over the limit.

File: callmeback/backend/src/test/java/org/google/callmeback/api/services/MovingAverageServiceTest.java (hunk 2 above)
Comment (egracer) [reply callmeback/backend/src/test/java/org/google/callmeback/api/services/MovingAverageServiceTest.java:129]: Done

File: callmeback/backend/src/test/java/org/google/callmeback/TestHelpers.java (hunk 3 above)
Comment (egracer) [reply callmeback/backend/src/test/java/org/google/callmeback/TestHelpers.java:11]: Done

**Pull request merged**